pymc/distributions/multivariate.py

```diff
@@ -2380,3 +2383,171 @@ def logp(value, alpha, K):
             K > 0,
             msg="alpha > 0, K > 0",
         )
+
+
+class ZeroSumNormalRV(SymbolicRandomVariable):
+    """ZeroSumNormal random variable"""
+
+    _print_name = ("ZeroSumNormal", "\\operatorname{ZeroSumNormal}")
+    zerosum_axes = None
+
+    def __init__(self, *args, zerosum_axes, **kwargs):
+        self.zerosum_axes = zerosum_axes
+        super().__init__(*args, **kwargs)
+
+
+class ZeroSumNormal(Distribution):
+    r"""
+    ZeroSumNormal distribution, i.e Normal distribution where one or
+    several axes are constrained to sum to zero.
+    By default, the last axis is constrained to sum to zero.
+    See `zerosum_axes` kwarg for more details.
+
+    Parameters
+    ----------
+    sigma : tensor_like of float
+        Standard deviation (sigma > 0).
+        Defaults to 1 if not specified.
+        For now, ``sigma`` has to be a scalar, to ensure the zero-sum constraint.
+    zerosum_axes: list or tuple of strings or integers
+        Axis (or axes) along which the zero-sum constraint is enforced.
+        Defaults to [-1], i.e the last axis.
+        If strings are passed, then ``dims`` is needed.
+        Otherwise, ``shape`` and ``size`` work as they do for other PyMC distributions.
+    dims: list or tuple of strings, optional
+        The dimension names of the axes.
+        Necessary when ``zerosum_axes`` is specified with strings.
+
+    Warnings
+    --------
+    ``sigma`` has to be a scalar, to ensure the zero-sum constraint.
+    The ability to specifiy a vector of ``sigma`` may be added in future versions.
+
+    Examples
+    --------
+    .. code-block:: python
+        COORDS = {
+            "regions": ["a", "b", "c"],
+            "answers": ["yes", "no", "whatever", "don't understand question"],
+        }
+        with pm.Model(coords=COORDS) as m:
+    ...:     v = pm.ZeroSumNormal("v", dims=("regions", "answers"), zerosum_axes="answers")
+
+        with pm.Model(coords=COORDS) as m:
+    ...:     v = pm.ZeroSumNormal("v", dims=("regions", "answers"), zerosum_axes=("regions", "answers"))
+
+        with pm.Model(coords=COORDS) as m:
+    ...:     v = pm.ZeroSumNormal("v", dims=("regions", "answers"), zerosum_axes=1)
+    """
+    rv_type = ZeroSumNormalRV
+
+    def __new__(cls, *args, zerosum_axes=None, dims=None, **kwargs):
+        dims = convert_dims(dims)
+        if zerosum_axes is None:
+            zerosum_axes = [-1]
+        if not isinstance(zerosum_axes, (list, tuple)):
+            zerosum_axes = [zerosum_axes]
+
+        if isinstance(zerosum_axes[0], str):
```

Comment:
Isn't that just a Normal distribution?

Reply:
It is, but if you write more general code that somehow produces the `zerosum_axes` automatically, that could easily happen I think.

Reply:
Oh so you mean erroring out in that case, gotcha

Reply:
Well, erroring out would be ok, but can't we just handle that case correctly? I can't think of anything that should go wrong in this case. So why not just 
```
if len(zerosum_axes) > 0 and isinstinace(zerosum_axes[0]), str):
```
And maybe add a test that checks if `pm.ZeroSumNormal("y", zerosum_axes=[], shape=(3,))` or so works and gives us a normal dist.

Reply:
Oh ok, I understand what you mean now. I'm curious what @ricardoV94 thinks, but I would prefer not to do that: if people wanna use a Normal they should just use `pm.Normal`, otherwise I find it makes code and models more confusing

Reply:
Actually I agree with you now @aseyboldt (mainly because that behavior would be consistent with other PyMC distributions' behavior). 
However, this edge-case needs a bit more work right now, because [`at.stack` in `get_support_shape`](https://github.com/pymc-devs/pymc/pull/6121/files#diff-00de3aa37024b2f780a8b60a86a006fbf3254e37c4aeb604eb3e38aaeb70df51R666) fails with an empty tuple.
As it's not a core feature of ZSN, I'm [marking this as TODO](https://github.com/pymc-devs/pymc/pull/6121/files#diff-ba7699f4e550fe6433c577626dd433a017c2fad177dcbfc83ea8268fe4322761R2491). The code is actually already there, so if someone needs this in the future, there are already some breadcrumbs to get started.